binding.cc

```diff
@@ -539,9 +545,16 @@ namespace zmq {
   Handle<Value> Socket::GetSockOpt(int option) {
     T value = 0;
     size_t len = sizeof(T);
-    if (zmq_getsockopt(socket_, option, &value, &len) < 0) {
-      NanThrowError(ExceptionFromError());
-      return NanUndefined();
+    while(true) {
+      int rc = zmq_getsockopt(socket_, option, &value, &len);
+      if ( rc < 0 && zmq_errno()==EINTR) {
+        continue;
+      } else if (rc<0) {
+        NanThrowError(ExceptionFromError());
+        return NanUndefined();
+      } else {
+        break;
+      }
```

Comment:
Would you mind equalizing the structure of this chunk to what you did from line 363-372? That way you only leave a single `rc < 0` comparison and keep some consistency.